addons/sale/models/sale_order_line.py

```diff
@@ -1249,6 +1249,15 @@ def write(self, values):
                       '\n'.join(fields.mapped('field_description')))
                 )
 
+        if 'product_id' in values and any(
+            sol.product_id.id != values['product_id']
+            and not sol.product_updatable
+            for sol in self
+        ):
```

Comment:
Shouldn't this be checked before editing any line? (see `_update_line_quantity` up there in the same method)

Reply:
Will be rolled back anyway, but I can move it earlier :ok_hand: